discord/role.py

```diff
@@ -317,6 +321,22 @@ def members(self) -> List[Member]:
         role_id = self.id
         return [member for member in all_members if member._roles.has(role_id)]
 
+    @property
+    def icon(self) -> Optional[Asset]:
+        """Optional[:class:`Asset`]: Returns the icon of the role. If role has no icon, then ``None`` will be returned.
+        
+        .. versionadded:: 2.0
+        """
+        if self._icon is None:
+            return None
+        
+        return Asset._from_icon(
+                self._state,
+                self.id,
+                self.icon,
```

Comment:
Is this correct? I think this would cause an Recursion Error

Reply:
Should've been `self._icon`. mb.